src/answers-search-bar.js

```diff
@@ -0,0 +1,519 @@
+   * either from a defaultInitialSearch or from a query in the URL.
+   */
+  _searchOnLoad () {
+    const searchComponents = this.components._activeComponents
+      .filter(c => c.constructor.type === SearchComponent.type);
+    if (searchComponents.length) {
+      searchComponents.forEach(c => c.searchAfterAnswersOnReady && c.searchAfterAnswersOnReady());
+    } else if (this.core.storage.has(StorageKeys.QUERY)) {
+      this.core.triggerSearch(this.core.storage.get(StorageKeys.QUERY_TRIGGER));
+    }
+  }
+
+  domReady (cb) {
+    DOM.onReady(cb);
+  }
+
+  onReady (cb) {
+    this._onReady = cb;
+    return this;
+  }
+
+  /**
+   * Parses the config provided by the user. In the parsed config, any options not supplied by the
+   * user are given default values.
+   * @param {Object} config The user supplied config.
+   */
+  parseConfig (config) {
+    const parsedConfig = Object.assign({}, DEFAULTS, config);
+    let sessionTrackingEnabled = true;
+    if (typeof config.sessionTrackingEnabled === 'boolean') {
+      sessionTrackingEnabled = config.sessionTrackingEnabled;
+    }
+    parsedConfig.sessionTrackingEnabled = sessionTrackingEnabled;
+
+    const sandboxPrefix = `${SANDBOX}-`;
+    parsedConfig.apiKey.includes(sandboxPrefix)
+      ? parsedConfig.environment = SANDBOX
+      : parsedConfig.environment = PRODUCTION;
+    parsedConfig.apiKey = parsedConfig.apiKey.replace(sandboxPrefix, '');
+
+    return parsedConfig;
+  }
+
+  /**
+   * Validates the Answers config object to ensure things like api key and experience key are
+   * properly set.
+   * @param {Object} config The Answers config.
+   */
+  validateConfig (config) {
+    // TODO (tmeyer): Extract this method into it's own class. Investigate the use of JSON schema
+    // to validate these configs.
+    if (typeof config.apiKey !== 'string') {
+      throw new Error('Missing required `apiKey`. Type must be {string}');
+    }
+
+    if (typeof config.experienceKey !== 'string') {
+      throw new Error('Missing required `experienceKey`. Type must be {string}');
+    }
+
+    if (config.onVerticalSearch && typeof config.onVerticalSearch !== 'function') {
+      throw new Error('onVerticalSearch must be a function. Current type is: ' + typeof config.onVerticalSearch);
+    }
+
+    if (config.onUniversalSearch && typeof config.onUniversalSearch !== 'function') {
+      throw new Error('onUniversalSearch must be a function. Current type is: ' + typeof config.onUniversalSearch);
+    }
+  }
+
+  addComponent (type, opts) {
+    if (typeof opts === 'string') {
+      opts = {
+        container: opts
+      };
+    }
+
+    try {
+      this.components.create(type, opts).mount();
+    } catch (e) {
+      throw new AnswersComponentError('Failed to add component', type, e);
+    }
+    return this;
+  }
+
+  /**
+   * Conducts a search in the Answers experience
+   *
+   * @param {string} query
+   */
+  search (query) {
```

Comment:
do we need ANSWERS.search or any of the below methods?

Reply:
I feel like we definitely don't need `search`, I'm meeting with Rose to decide on the other methods.

Reply:
It's unclear to me how conversions/sessions opt-ins would/should work on a redirect search bar.